frontend/src/pages/AccountSend.vue

```diff
@@ -469,15 +815,22 @@ function useSendForm() {
     return humanizeMinSendAmount(minSend);
   };
 
-  function isValidTokenAmount(val: string | undefined) {
+  function isValidTokenAmount(val: string | undefined, tokenInput?: TokenInfoExtended | null | undefined) {
     if (val === undefined) return true; // don't show error on empty field
     if (!val || !(Number(val) > 0)) return tc('Send.enter-an-amount');
-    if (!token.value) return tc('Send.select-a-token');
 
-    const { address: tokenAddress, decimals } = token.value;
+    let tokenToUse: TokenInfoExtended | null | undefined;
+    // Use batchSend token if it exists
+    if (tokenInput) tokenToUse = tokenInput;
+    // Otherwise use the current Send token
+    else tokenToUse = token.value;
+    if (!tokenToUse) return tc('Send.select-a-token');
+
+    const { address: tokenAddress, decimals } = tokenToUse;
+
     const minAmt = getMinSendAmount(tokenAddress);
     if (Number(val) < minAmt && isNativeToken(tokenAddress)) return `${tc('Send.send-at-least')} ${minAmt} ${NATIVE_TOKEN.value.symbol}`; // prettier-ignore
-    if (Number(val) < minAmt && !isNativeToken(tokenAddress)) return `${tc('Send.send-at-least')} ${minAmt} ${token.value.symbol}`; // prettier-ignore
+    if (Number(val) < minAmt && !isNativeToken(tokenAddress)) return `${tc('Send.send-at-least')} ${minAmt} ${tokenToUse.symbol}`; // prettier-ignore
```

Comment:
Only one of these can be true, so `else if` seems more clear
```suggestion
    if (Number(val) < minAmt && isNativeToken(tokenAddress)) return `${tc('Send.send-at-least')} ${minAmt} ${NATIVE_TOKEN.value.symbol}`; // prettier-ignore
    else if (Number(val) < minAmt && !isNativeToken(tokenAddress)) return `${tc('Send.send-at-least')} ${minAmt} ${tokenToUse.symbol}`; // prettier-ignore
```

Reply:
Think my suggestion got messed up here by highlighting too many lines but I think you get the idea

Reply:
Like this? https://github.com/ScopeLift/umbra-protocol/pull/491/commits/7ac43663c8820bb6153315566b17291a95d40114